File: configlibrary/src/main/java/com/shmedo/configlibrary/ble/enums/LowEnergyModel.java
package com.shmedo.configlibrary.ble.enums;

/**
 * Created by adu on 2017/12/18.
 * DAS低功耗模式
 */
public enum LowEnergyModel {
    //待机
    STANDBY(1),
    //激活
    ACTIVATE(2);

    private int model;

    LowEnergyModel(int model) {
        this.model = model;
    }


    public int toInt() {
        return model;
    }

    public static LowEnergyModel valueOf(int model) {
        switch (model) {
            case 1:
                return STANDBY;

            case 2:
                return ACTIVATE;

            default:
                return STANDBY;
        }
    }
}

File: lib_device_base/src/main/java/com/shmedo/lib/device/base/md_cmd/enums/LowEnergyModel.kt
package com.shmedo.lib.device.base.md_cmd.enums

/**
 * 创建者：gonghe
 * 创建时间：2024/4/10
 * 描述：  DAS低功耗模式
 */
enum class LowEnergyModel(private val model: Int) {

    STANDBY(1),  //待机
    ACTIVATE(2); //激活

    fun toInt(): Int {
        return model
    }

    companion object {
        @JvmStatic
        fun value(port: Int) = when (port) {
            1 -> STANDBY
            else -> ACTIVATE
        }
    }
}